File: app/src/main/java/com/zxp/mvpcuoqv/view/MainActivity.java
package com.zxp.mvpcuoqv.view;

import android.widget.Toast;

import androidx.recyclerview.widget.RecyclerView;

import com.scwang.smartrefresh.layout.SmartRefreshLayout;
import com.zxp.data.Databean;
import com.zxp.frame.ApiConfig;
import com.zxp.frame.IContractModel;
import com.zxp.frame.IContractView;
import com.zxp.frame.LoadTypeConfig;
import com.zxp.mvpcuoqv.model.TestModel;
import com.zxp.frame.PrameHashMap;
import com.zxp.mvpcuoqv.R;
import com.zxp.mvpcuoqv.adapter.TestAdapter;
import com.zxp.mvpcuoqv.base.BaseMvpActivity;
import com.zxp.mvpcuoqv.interfaces.DataInterfaces;

import java.util.List;
import java.util.Map;



public class MainActivity extends BaseMvpActivity implements IContractView {

/*    @BindView(R.id.recy_view)
    RecyclerView recyView;
    @BindView(R.id.RefreshLayout)
    SmartRefreshLayout RefreshLayout;*/
    private int pageId = 0;
    private Map<String, Object> prameHashMap;
    private TestAdapter testAdapter;
    private SmartRefreshLayout RefreshLayout;
    private RecyclerView recy_view;


    @Override
    protected int getlayout() {
        return R.layout.activity_main;
    }

    @Override
    public IContractModel setModel() {
        return new TestModel();
    }

    @Override
    public void setUpView() {
        prameHashMap = new PrameHashMap().add("c", "api").add("a", "getList");
        recy_view = findViewById(R.id.recy_view);
        RefreshLayout = findViewById(R.id.RefreshLayout);
        initRecyclerView(recy_view,RefreshLayout,  new DataInterfaces() {
            @Override
            public void dataType(int mode) {
                if (mode==LoadTypeConfig.REFRESH){
                    pageId++;
                    contractPersenter.getData(ApiConfig.TEST_GET, LoadTypeConfig.REFRESH, prameHashMap, pageId);

                }else {
                    pageId = 0;
                    contractPersenter.getData(ApiConfig.TEST_GET, LoadTypeConfig.MORE, prameHashMap, pageId);
                }
            }
        });

        testAdapter = new TestAdapter(this);
        recy_view.setAdapter(testAdapter);

    }

    @Override
    public void setUpData() {

        contractPersenter.getData(ApiConfig.TEST_GET, LoadTypeConfig.NORMAL, prameHashMap, pageId);
    }

    @Override
    public void netSuccess(int whichApi, Object[] pa) {
        switch (whichApi) {
            case ApiConfig.TEST_GET:
                if ((int)pa[0] == LoadTypeConfig.MORE) {
                    RefreshLayout.finishLoadMore();
                } else if ((int)pa[0] == LoadTypeConfig.REFRESH) {
                    RefreshLayout.finishRefresh();
                    testAdapter.clear();
                }
                List<Databean.DatasBean> datas = ((Databean) pa[0]).getDatas();
                testAdapter.add(datas);
                break;
        }
    }


    @Override
    public void onData(int whichApi, Object[] pa) {

    }

    @Override
    public void error(int which, Throwable throwable) {
        Toast.makeText(MainActivity.this, throwable.getMessage() != null ? throwable.getMessage() : "网络请求发生错误", Toast.LENGTH_SHORT).show();
    }

}

File: xiaomokong/src/main/java/com/zxp/xiaomokong/MainActivity.kt
package com.zxp.xiaomokong

import com.zxp.data.DemoData
import com.zxp.data.DemoKotlinInfo
import com.zxp.frame.ApiConfigKtx
import com.zxp.frame.LoadTypeConfig
import com.zxp.xiaomokong.adapter.DemoAdapter
import com.zxp.xiaomokong.base.BaseMvpActivityKt
import com.zxp.xiaomokong.interfacse.DataInterfacesKt
import com.zxp.xiaomokong.model.DemoModel
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseMvpActivityKt<DemoModel>(), DataInterfacesKt {
    override fun setModel(): DemoModel = DemoModel()
    private var adapter: DemoAdapter? = null
    private var pageNum = 0
    var mListData: MutableList<DemoData> = mutableListOf()
    var map: MutableMap<String, Any>?=null;

    override fun setLayoutId(): Int = R.layout.activity_main


    override fun setUpView() {
        mPresenter.allowLoading(this)
        initRecyclerView(recyclerView, refreshLayout, this)
        adapter = recyclerView.let {
            var adapter1 = DemoAdapter(mListData, this)
            it.adapter = adapter1
            adapter1
        }
    }

    override fun setUpData() {
         map = mutableMapOf(Pair("c", "api"), Pair("a", "getList"), Pair("page_id", pageNum))
        mPresenter.goNetWork(ApiConfigKtx.SPLASH, LoadTypeConfig.NORMAL, map!!, 6)
    }

    override fun onNetSuccess(whichApi: Int, loadType: Int, backContent: Array<Any>) {
        when (whichApi) {
            ApiConfigKtx.SPLASH -> {
                if (loadType == LoadTypeConfig.MORE) refreshLayout.finishLoadMore()
                else if (loadType == LoadTypeConfig.REFRESH) {
                    refreshLayout.finishRefresh()
                    mListData.clear()
                }
                var info: DemoKotlinInfo = backContent[0] as DemoKotlinInfo
                mListData.addAll(info.datas)
                adapter!!.notifyDataSetChanged()
            }
        }
    }

    override fun dataType(mode: Int) {
        if (mode == LoadTypeConfig.REFRESH) {
            pageNum = 0
            mPresenter.goNetWork(ApiConfigKtx.SPLASH, LoadTypeConfig.REFRESH, map!!, 6)
        } else if (mode == LoadTypeConfig.MORE) {
            pageNum += 1
            mPresenter.goNetWork(ApiConfigKtx.SPLASH, LoadTypeConfig.MORE, map!!, 6)
        }

    }

}
